File: src/main/java/com/libreria/sistema/service/UsuarioService.java
package com.libreria.sistema.service;

import com.libreria.sistema.dto.request.UsuarioRequest;
import com.libreria.sistema.dto.response.UsuarioResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface UsuarioService {

    Page<UsuarioResponse> listar(String busqueda, Pageable pageable);

    UsuarioResponse buscarPorId(Long id);

    UsuarioResponse crear(UsuarioRequest request);

    UsuarioResponse actualizar(Long id, UsuarioRequest request);

    void eliminar(Long id);

    void cambiarEstado(Long id, boolean estado);
}

File: frontend/js/services/usuarioService.js
import { api } from '../utils.js';

const UsuarioService = {
  listar: (params) => api(`/api/usuarios?${new URLSearchParams(params)}`),
  buscarPorId: (id) => api(`/api/usuarios/${id}`),
  crear: (data) => api('/api/usuarios', { method: 'POST', body: JSON.stringify(data) }),
  actualizar: (id, data) => api(`/api/usuarios/${id}`, { method: 'PUT', body: JSON.stringify(data) }),
  eliminar: (id) => api(`/api/usuarios/${id}`, { method: 'DELETE' }),
  cambiarEstado: (id, estado) => api(`/api/usuarios/${id}/estado?estado=${estado}`, { method: 'PATCH' }),
};

export default UsuarioService;
